src/sentry/issues/auto_source_code_config/derived_code_mappings_endpoint.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from typing import Literal
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.serializers import serialize
+from sentry.integrations.source_code_management.repo_trees import (
+    RepoAndBranch,
+    RepoTreesIntegration,
+)
+from sentry.issues.auto_source_code_config.integration_utils import (
+    InstallationCannotGetTreesError,
+    InstallationNotFoundError,
+)
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+
+from .code_mapping import (
+    CodeMapping,
+    CodeMappingTreesHelper,
+    FrameInfo,
+    NeedsExtension,
+    create_code_mapping,
+)
+from .integration_utils import get_installation
+
+logger = logging.getLogger(__name__)
+
+
+def process_get_request(request: Request, organization: Organization) -> Response:
+    try:
+        file_repo_matches = []
+        resp_status: Literal[200, 204, 400] = status.HTTP_400_BAD_REQUEST
+
+        file_repo_matches = get_file_and_repo_matches(request, organization)
+        if file_repo_matches:
+            resp_status = status.HTTP_200_OK
+        else:
+            resp_status = status.HTTP_204_NO_CONTENT
+
+        return Response(serialize(file_repo_matches), status=resp_status)
+    except InstallationCannotGetTreesError:
+        return Response(
+            {"text": "The integration does not support getting trees"},
+            status=status.HTTP_404_NOT_FOUND,
+        )
+    except InstallationNotFoundError:
+        return Response(
+            {"text": "Could not find this integration installed on your organization"},
+            status=status.HTTP_404_NOT_FOUND,
+        )
+    except NeedsExtension:
+        return Response({"text": "Needs extension"}, status=status.HTTP_400_BAD_REQUEST)
+    except KeyError:
+        return Response({"text": "Missing required parameters"}, status=status.HTTP_400_BAD_REQUEST)
+
+
+def get_file_and_repo_matches(request: Request, organization: Organization) -> list[dict[str, str]]:
+    frame_info = get_frame_info_from_request(request)
+    installation = get_installation(organization)
+    if not isinstance(installation, RepoTreesIntegration):
+        return []
+    trees = installation.get_trees_for_org()
+    trees_helper = CodeMappingTreesHelper(trees)
+    return trees_helper.get_file_and_repo_matches(frame_info)
+
+
+def get_frame_info_from_request(request: Request) -> FrameInfo:
+    frame = {
+        "absPath": request.GET.get("absPath"),
+        # Currently, the only required parameter
+        "filename": request.GET.get("stacktraceFilename"),
```

Comment:
The frontend currently only sends `filename`, however, we're going to need the other two values as well.